Modules/_localemodule.c

```diff
@@ -623,6 +623,45 @@ _locale_nl_langinfo_impl(PyObject *module, int item)
 }
 #endif /* HAVE_LANGINFO_H */
 
+PyDoc_STRVAR(getfirstweekday__doc__,
+"getfirstweekday() -> integer\n"
+"Return the first day of week. 0 is Monday, 6 is Sunday and None is unknown");
+
+static PyObject*
+PyLocale_getfirstweekday(PyObject* self, PyObject* args)
+{
+    int start = -1;
+#if defined(MS_WINDOWS)
+    char locale[1];
+
+    if (!GetLocaleInfo(LOCALE_USER_DEFAULT,
+                       LOCALE_IFIRSTDAYOFWEEK,
+                       locale, sizeof(locale))) {
+        start = (locale[0] - '0') % 7;
+    }
+#else
+#ifdef HAVE__NL_TIME_FIRST_WEEKDAY
+    int first_weekday = nl_langinfo(_NL_TIME_FIRST_WEEKDAY)[0];
+    long week_1stday = (long)nl_langinfo(_NL_TIME_WEEK_1STDAY);
+    switch (week_1stday) {
+    default:
+    case 19971130:
```

Comment:
What is the meaning of this magic constant?

Reply:
They come from LC_TIME week from locale: https://man7.org/linux/man-pages/man5/locale.5.html

Reply:
Please add a comment where the magic constant is coming from and that it is a glibc specific magic constant.